### PR TITLE
Limit missing primary key fail to new tables and comment out for now

### PR DESCRIPTION
**Pull request by nickvergessen**
This will work on CI so devs notice it when they install the app for testing,
and at the same time existing faulty tables don't break the upgrade to 24

Regression from #31513

### CODE REVIEW

**Review by ChristophWurst — APPROVED** [review]

**Comment by nickvergessen** [comment]
Reasons to disable for now:
* occ db:add-missing-primarykeys can't fix it before the upgrade is run
* e2ee fails
* circles fails

**Comment by nickvergessen** [comment]
cc @PVince81 lets disable for now

**Review by PVince81 — APPROVED** [review]
:+1:

File: lib/private/DB/MigrationService.php [hunk 1]
```diff
@@ -634,8 +634,8 @@ public function ensureOracleConstraints(Schema $sourceSchema, Schema $targetSche
 				if ($isUsingDefaultName && \strlen($table->getName()) - $prefixLength >= 23) {
 					throw new \InvalidArgumentException('Primary index name on "' . $table->getName() . '" is too long.');
 				}
-			} elseif (!$primaryKey instanceof Index) {
-				throw new \InvalidArgumentException('Table "' . $table->getName() . '" has no primary key and therefor will not behave sane in clustered setups.');
+			// } elseif (!$primaryKey instanceof Index && !$sourceTable instanceof Table) {
+				// throw new \InvalidArgumentException('Table "' . $table->getName() . '" has no primary key and therefor will not behave sane in clustered setups.');
```
Comment (PVince81) [inline lib/private/DB/MigrationService.php:638]: can we log a warning instead ?

Comment (nickvergessen) [reply lib/private/DB/MigrationService.php:638]: We don't have the logger in here yet. Let's do later.

**Comment by nickvergessen** [comment]
Rebased as the PR was broken after Carl rebased when github had issues last week

**Pull request merged**